autopush-common/src/reliability.rs

```diff
@@ -363,16 +382,18 @@ impl PushReliability {
                 // it should return `nil`, in which case we record a soft error.
                 // This could also be strung together as a cascade of functions, but it's broken
                 // out to discrete steps for readability.
+                if result == redis::Value::Nil {
+                    warn!("🔍⚠🪈 {id} - Pipe failed, retry.");
+                    return Ok(None);
+                }
                 if let Some(operations) = result.as_sequence() {
```

Comment:
Now that this set is conditional on `!= Received` shouldn't the handling of the result be as well? (with the last operation being hincr instead of set)

Reply:
No. In this case, a `Nil` response would indicate that the pipe failed. Otherwise, it would return a structured response that contains the result of the queued operation.

Redis has horrible error handling.

Reply:
I mean post this first `result == redis::Value::Nil {` check -- we're looking at the `last()` of the sequence, which would in the case of Received be the `hincr`

Reply:
Well, while the operation has changed, the result is still functionally the same. If there is a problem with the Pipe, we would get back a `Nil`. If the pipe succeeded, we would get back the adjusted value. if the pipe failed, we would still want to note it as an error. 

So the first check is to see if the lock failed and returned Nil. The second check is to see if the internal pipe failed and returned Nil.

Reply:
Sorry, I've made this entirely too confusing (at least to myself) by mistakenly commenting on the wrong line.. Let me try again on line 396 just so we're on the same page..